blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubPipelineCreateRequest.java

```diff
@@ -196,6 +204,23 @@ public boolean meetsIndexingCriteria() {
         return null;
     }
 
+    private void updateEndpoints(String apiUrl) {
+        GitHubConfiguration config = GitHubConfiguration.get();
+        synchronized (config) {
+            final String finalApiUrl = apiUrl;
+            Endpoint endpoint = Iterables.find(config.getEndpoints(), new Predicate<Endpoint>() {
+                @Override
+                public boolean apply(@Nullable Endpoint input) {
+                    return input != null && input.getApiUri().equals(finalApiUrl);
+                }
+            }, null);
+            if (endpoint == null) {
+                config.setEndpoints(ImmutableList.of(new Endpoint(apiUrl, apiUrl)));
+                config.save();
+            }
+        }
```

Comment:
@i386 @vivek just for my own understanding, you need to add the endpoint URL to the branch source `GitHubConfiguration` object? Otherwise it won't allow for a custom endpoint to be used?